app/src/main/java/com/practicum/playlistmaker/App.kt

```diff
@@ -33,9 +34,18 @@ class App : Application() {
             SettingsLocalDataSourceImpl(getSharedPreferences(PREFS_NAME, MODE_PRIVATE))
         settingsRepository = SettingsRepositoryImpl(settingsStorage)
 
-        switchTheme(
+        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
+
+        val sharedPreferences = getSharedPreferences(PREFS_NAME, MODE_PRIVATE)
+        val darkThemeEnabled = if (sharedPreferences.contains(DARK_THEME_KEY)) {
             settingsRepository.getThemeSettings().isNightModeEnabled
-        )
+        } else {
+            null
+        }
+
+        if (darkThemeEnabled != null) {
+            switchTheme(darkThemeEnabled)
+        }
```

Comment:
🍏  Можно еще такой вариант использовать:
```suggestion
        darkThemeEnabled?.let {
            switchTheme(darkThemeEnabled)
        }
```